src/DefaultBuilder/src/ConfigureWebHostBuilder.cs

```diff
@@ -50,9 +50,8 @@ public IWebHostBuilder ConfigureAppConfiguration(Action<WebHostBuilderContext, I
 
             if (_configuration[WebHostDefaults.WebRootKey] is string value && !string.Equals(previousWebRoot, value, StringComparison.OrdinalIgnoreCase))
             {
-                // We allow changing the web root since it's based off the content root and typically
-                // read after the host is built.
-                _environment.WebRootPath = Path.Combine(_environment.ContentRootPath, value);
+                // Diasllow changing the web root for consistency with other types.
+                throw new NotSupportedException($"The application name changed from \"{previousWebRoot}\" to \"{_configuration[WebHostDefaults.WebRootKey]}\". Changing the host configuration using WebApplicationBuilder.WebHost is not supported. Use WebApplication.CreateBuilder(WebApplicationOptions) instead.");
```

Comment:
Is "application name" the right name here?  Also, shouldn't the message say that `WebApplicationBuilder.UseWebHost` is not supported, instead of `WebApplicationBuilder.WebHost`

Reply:
Copy pasta

Reply:
@captainsafia we also need to normalize the web root

Reply:
Good catches on both counts! Resolved.